corehq/apps/sso/views.py

```diff
@@ -0,0 +1,159 @@
+        request.session['samlNameIdSPNameQualifier'] = request.saml2_auth.get_nameid_spnq()
+        request.session['samlSessionIndex'] = request.saml2_auth.get_session_index()
+
+        # todo for debugging purposes to dump into the response below
+        request_session_data = {
+            "samlUserdata": request.session['samlUserdata'],
+            "samlNameId": request.session['samlNameId'],
+            "samlNameIdFormat": request.session['samlNameIdFormat'],
+            "samlNameIdNameQualifier": request.session['samlNameIdNameQualifier'],
+            "samlNameIdSPNameQualifier": request.session['samlNameIdSPNameQualifier'],
+            "samlSessionIndex": request.session['samlSessionIndex'],
+        }
+
+        # todo redirect here?
+        saml_relay = OneLogin_Saml2_Utils.get_self_url(request.saml2_request_data)
+
+        # todo this is the point where we would initiate a django auth session
+
+    else:
+        error_reason = request.saml2_auth.get_last_error_reason()
+
+    return HttpResponse(json.dumps({
+        "errors": errors,
+        "error_reason": error_reason,
+        "not_auth_warn": not_auth_warn,
+        "request_id": request_id,
+        "processed_response": processed_response,
+        "saml_relay": saml_relay,
+        "request_session_data": request_session_data,
+    }), 'text/json')
+
+
+@use_saml2_auth
+def sso_saml_sls(request, idp_slug):
+    """
+    SLS stands for Single Logout Service. This view is responsible for
+    handling a logout response from the Identity Provider.
+    """
+    # todo these are placeholders for the json dump below
+    error_reason = None
+    success_slo = False
+    attributes = False
+    saml_user_data_present = False
+
+    request_id = request.session.get('LogoutRequestID')
+    url = request.saml2_auth.process_slo(
+        request_id=request_id,
+        delete_session_cb=lambda: request.session.flush()
+    )
+    errors = request.saml2_auth.get_errors()
+
+    if len(errors) == 0:
+        if url is not None:
+            return HttpResponseRedirect(url)
+        else:
+            success_slo = True
+    elif request.saml2_auth.get_settings().is_debug_active():
+        error_reason = request.saml2_auth.get_last_error_reason()
+
+    # todo what's below is a debugging placeholder
+    if 'samlUserdata' in request.session:
+        saml_user_data_present = True
+        if len(request.session['samlUserdata']) > 0:
+            attributes = request.session['samlUserdata'].items()
+
+    return HttpResponse(json.dumps({
+        "errors": errors,
+        "error_reason": error_reason,
+        "success_slo": success_slo,
+        "attributes": attributes,
+        "saml_user_data_present": saml_user_data_present,
+    }), 'text/json')
+
+
+@use_saml2_auth
+def sso_saml_login(request, idp_slug):
+    """
+    This view initiates a SAML 2.0 login request with the Identity Provider.
+    """
+    return HttpResponseRedirect(request.saml2_auth.login())
```

Comment:
does this gracefully handle failure?

Reply:
no, that isn't the goal here as more things might change when I actually integrate this with auth. this is just the initial views